File: frontend/src/types/detection.ts
export interface DetectionItem {
  id: number
  label: string
  confidence: number
  x_min: number
  y_min: number
  x_max: number
  y_max: number
}

export interface Detection {
  id: number
  original_filename: string
  annotated_image_path: string
  person_count: number
  status: string
  created_at: string
  items: DetectionItem[]
}

export interface DetectionListItem {
  id: number
  original_filename: string
  person_count: number
  status: string
  created_at: string
}

export interface HealthResponse {
  status: string
  app_name: string
}

File: backend/app/schemas/detection.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DetectionItemResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    label: str
    confidence: float
    x_min: float
    y_min: float
    x_max: float
    y_max: float


class DetectionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    original_filename: str
    annotated_image_path: str
    person_count: int
    status: str
    created_at: datetime
    items: list[DetectionItemResponse] = []


class DetectionListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    original_filename: str
    person_count: int
    status: str
    created_at: datetime
